datasetinsights/estimators/faster_rcnn.py

```diff
@@ -355,619 +884,123 @@ def _gt_preds2tensor(gt_preds, max_boxes=MAX_BOXES_PER_IMAGE):
-            )
-        train_sampler = self._create_sampler(
-            dataset=train_dataset, is_train=True,
-        )
-        val_sampler = self._create_sampler(dataset=val_dataset, is_train=False)
+    Returns (list(tuple(list(BBox2d), (Bbox2d)))): A list of tuples where
+        the first element in each tuple is a list of bounding boxes
+        corresponding to the targets in an example, and the second element
+        in the tuple corresponds to the predictions in that example
 
-        train_loader = self._create_loader(
-            dataset=train_dataset,
-            batch_size=self.config.train.batch_size,
-            sampler=train_sampler,
-            collate_fn=collate_fn,
-            train=True,
-        )
-        val_loader = self._create_loader(
-            dataset=val_dataset,
-            batch_size=self.config.val.batch_size,
-            sampler=val_sampler,
-            collate_fn=collate_fn,
-            train=False,
-        )
-        self.train_loop(
-            config=self.config,
-            train_dataloader=train_loader,
-            label_mappings=label_mappings,
-            val_dataloader=val_loader,
-            writer=self.writer,
-            kfp_writer=self.kfp_writer,
-            model=self.model,
-            train_sampler=train_sampler,
-        )
+    """
+    gt_preds_canonical = []
+    for t in tensor_list:
+        gt_preds_canonical += list3d_2canonical(t.tolist())
+    return gt_preds_canonical
 
-    def save(self, path):
-        """ Serialize Estimator to path
 
-        Args:
-            path (str): full path to save serialized estimator
+def metric_per_class_plot(metric_name, data, label_mappings, figsize=(20, 10)):
+    """
+    Bar plot for metric per class.
+    Args:
+        metric_name (str): metric name.
+        data (dict): a dictionary of metric per label.
+        label_mappings (dict): a dict of {label_id: label_name} mapping
+        figsize (tuple): figure size of the plot. Default is (20, 10)
 
-        Returns:
-            saved full path of the serialized estimator
-        """
-        save_dict = {
-            "model": self.model_without_ddp.state_dict(),
-            "config": self.config,
-        }
-        torch.save(save_dict, path)
+    Returns (matplotlib.pyplot.figure): a bar plot for metric per class.
+    """
+    label_id = [k for k in sorted(data.keys()) if k in label_mappings]
+    metric_values = [data[i] for i in label_id]
+    label_name = [label_mappings[i] for i in label_id]
+    fig = plt.figure(figsize=figsize)
+    plt.bar(label_id, metric_values)
+    plt.title(f"{metric_name} per class")
+    plt.xlabel("label name")
+    plt.ylabel(f"{metric_name}")
+    plt.xticks(
+        label_id, label_name, rotation="vertical",
+    )
+    plt.margins(0.2)
+    plt.subplots_adjust(bottom=0.3)
+    return fig
 
-        return path
 
-    def load(self, path):
-        """ Load Estimator from path
+class PrepBoxes:
```

Comment:
This seems to be an opportunity combining `ToTensor` and `PrepBoxes`. One is responsible of converting PIL image to tensor the other is responsible of converting BBox2D objects to tensor.

Reply:
why to combine both? both have different purpose? and it has call implementation, which does the object content transformation.

Reply:
Maybe this calls for renaming PrepBoxes? Can it be more explicit? I am OK if we have steps that handle bounding boxes in a format separate from ToTensor code, but then I will argue why is `ToTensor` method not responsible for converting from bounding box is a certain format to pytorch tensor? With this design, maybe we can simple replace `ToTensor` with `ToTFTensor` so that we can support this method for tensorflow?